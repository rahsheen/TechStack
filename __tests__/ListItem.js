import 'react-native';
import React from 'react';
import { ListItem } from '../src/components/ListItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const mockLibrary = {
        "id": 2,
        "title": "Redux",
        "description": "Redux is a predictable state container for JavaScript apps. It helps you write applications that behave consistently, run in different environments (client, server, and native), and are easy to test."
    }

test('renders correctly', () => {
  const tree = renderer.create(
      <ListItem library={mockLibrary} />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})