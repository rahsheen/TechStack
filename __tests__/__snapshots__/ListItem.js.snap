// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={undefined}
  testID={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 5,
        "position": "relative",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "paddingLeft": 15,
        }
      }
    >
      Redux
    </Text>
  </View>
</View>
`;
